sensor-validate.cpp

```diff
@@ -1,28 +1,44 @@
 #include "sensor-validate.h"
 
-bool _give_me_a_good_name(double value, double nextValue, double maxDelta) {
-  if(nextValue - value > maxDelta) {
+bool isSensorReadingsHaveSuddenJumps_b(double currentValue, double nextValue, double maxDelta) {
+  if((nextValue - currentValue) > maxDelta) {
     return false;
   }
   return true;
 }
 
-bool validateSOCreadings(double* values, int numOfValues) {
-  int lastButOneIndex = numOfValues - 1;
-  for(int i = 0; i < lastButOneIndex; i++) {
-    if(!_give_me_a_good_name(values[i], values[i + 1], 0.05)) {
-      return false;
+bool getReadingStatus_b(double* values, int numOfReadings, double maxDelta)
+{
+   for(int ReadingIndex = 0; ReadingIndex < (numOfReadings -1); ReadingIndex++) 
+    {
+      if(!isSensorReadingsHaveSuddenJumps_b(values[ReadingIndex], values[ReadingIndex + 1], maxDelta)) 
+      {
+        return false;
+      }
     }
-  }
   return true;
 }
 
-bool validateCurrentreadings(double* values, int numOfValues) {
-  int lastButOneIndex = numOfValues - 1;
-  for(int i = 0; i < lastButOneIndex; i++) {
-    if(!_give_me_a_good_name(values[i], values[i + 1], 0.1)) {
-      return false;
-    }
+bool validateSOCreadings_b(double* values, int numOfReadings) 
+{
+  if (numOfReadings == 0)
+  {
+    return false;
+  }
+  else
+  {
+    return getReadingStatus_b(values, numOfReadings, 0.05);
+  }
+}
+
+bool validateCurrentreadings_b(double* values, int numOfReadings) 
+{
+  if (numOfReadings == 0)
```

Comment:
Repetition of input value check. If anything needs to be added here (like checking individual values or the number of readings), we will need to do it in two places.

Reply:
Hi @sudeeprp , Updated the test validation file and source files to avoid repetition , Please review it once again. Thank You :)